app/abci.go

```diff
@@ -812,3 +826,45 @@ func (app *HeimdallApp) PreBlocker(ctx sdk.Context, req *abci.RequestFinalizeBlo
 
 	return app.ModuleManager.PreBlock(ctx)
 }
+
+func (app *HeimdallApp) getInitialHeight(ctx sdk.Context) (int64, error) {
+	chain := helper.GetConfig().Chain
+	initialHeight := int64(0)
+	var err error
+
+	// this is for UTs that have non zero initial height
+	if chain != helper.MainChain && chain != helper.AmoyChain && chain != helper.MumbaiChain {
+		initialHeight, err = app.ChainManagerKeeper.GetInitialChainHeight(ctx)
+		if err != nil {
+			return -1, fmt.Errorf("error occurred while getting initial chain height: %w", err)
```

Comment:
So this is called by `getValidatorSetForHeight` and calling points are the ABCI methods. Wondering if it's better to return 0 and no error in case (to avoid panics).